prov/verbs/src/verbs_eq.c

```diff
@@ -766,6 +833,24 @@ fi_ibv_eq_read(struct fid_eq *eq_fid, uint32_t *event,
 	if ((ret = fi_ibv_eq_read_event(eq, event, buf, len, flags)))
 		return ret;
 
+	if (domain) {
+		/* Non-blocking call to retrieve IB async events */
+		if (fi_ibv_read_async_event(domain, eq)) {
+			return -FI_EAVAIL;
+		}
+	} else {
+		fastlock_acquire(&eq->lock);
+		dlist_foreach_container(&eq->domain_list, struct fi_ibv_domain,
+		                        domain, eq_dlentry) {
+			/* Non-blocking call to retrieve IB async events */
+			if (fi_ibv_read_async_event(domain, eq)) {
+				fastlock_release(&eq->lock);
+				return -FI_EAVAIL;
+			}
+		}
```

Comment:
I think we can use `epoll_wait` with timeout of 0 to still have it non-blocking and get the correct domain through `event.data.ptr`